superset-frontend/src/components/Select/AsyncSelect.tsx

```diff
@@ -425,8 +436,51 @@ const AsyncSelect = forwardRef(
       if (onClear) {
         onClear();
       }
+      fireOnChange();
+    };
+
+    const handleOnBlur = (event: React.FocusEvent<HTMLElement>) => {
+      const tagsMode = !isSingleMode && allowNewOptions;
+      const searchValue = inputValue.trim();
+      // Searched values will be autoselected during onBlur events when in tags mode.
+      // We want to make sure a value is only selected if the user has actually selected it
+      // by pressing Enter or clicking on it.
+      if (
+        tagsMode &&
+        searchValue &&
+        !hasOption(searchValue, selectValue, true)
+      ) {
+        // The search value will be added so we revert to the previous value
+        setSelectValue(selectValue || []);
+      }
+      onBlur?.(event);
     };
 
+    useEffect(() => {
+      if (onChangeCount !== previousChangeCount) {
+        const set = new Set();
+        const array = ensureIsArray(selectValue);
+        array.forEach(item => set.add(getValue(item)));
+        const options = mapOptions(
+          fullSelectOptions.filter(opt => set.has(opt.value)),
+        );
+        if (isSingleMode) {
+          // @ts-ignore
```

Comment:
The original typescript definition does not allow `undefined` but I checked and the component allows it when using `allowClear`.